semantic-convention-utils/src/main/java/org/hypertrace/semantic/convention/utils/db/DbSemanticConventionUtils.java

```diff
@@ -154,9 +170,65 @@ public static String getOperationFromDbQuery(String query) {
     return Optional.empty();
   }
 
-  public static String getDbOperationForMongo(Event event) {
-    return SpanAttributeUtils.getFirstAvailableStringAttribute(
-        event, DbSemanticConventionUtils.getAttributeKeysForMongoOperation());
+  public static Optional<String> getDbOperationForMongo(Event event) {
+    return Optional.ofNullable(
+        SpanAttributeUtils.getFirstAvailableStringAttribute(
+            event, DbSemanticConventionUtils.getAttributeKeysForMongoOperation()));
+  }
+
+  public static Optional<String> getDbName(Event event) {
+    return Optional.ofNullable(
+        SpanAttributeUtils.getFirstAvailableStringAttribute(event, getAttributeKeysForDbName()));
+  }
+
+  public static Optional<String> getSqlTableName(Event event) {
+    return Optional.ofNullable(
+        SpanAttributeUtils.getFirstAvailableStringAttribute(
+            event, getAttributeKeysForSqlTableName()));
+  }
+
+  public static Optional<String> getMongoCollectionName(Event event) {
+    return Optional.ofNullable(
+        SpanAttributeUtils.getFirstAvailableStringAttribute(
+            event, getAttributeKeysForMongoNamespace()));
+  }
+
+  public static Optional<String> getRedisDbIndex(Event event) {
+    return Optional.ofNullable(
+        SpanAttributeUtils.getFirstAvailableStringAttribute(
+            event, getAttributeKeysForRedisTableIndex()));
+  }
+
+  public static Optional<String> getDestinationForDb(Event event, Optional<String> table_name) {
```

Comment:
We prefer using camelCase for argument/variable/function names

`s/table_name/tableName`

Reply:
Can this function be made private?